Week 6/Subtree/Solution.java

```diff
@@ -0,0 +1,81 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+	//Node structure considered for this problem
+	class Node {
+		int value;
+		Node left;
+		Node right;
+
+		Node(int value, Node left,Node right){
+			this.value = value;
+			this.left = left;
+			this.right = right;
+		}
+	}
+
+    public boolean isSubtree(Node t, Node s) {
+    	//input validation
+    	if(t == null) return true;
+    	if(s == null) return false;
+    	//queue for BFS
+    	Queue<Node> bfs = new LinkedList<Node>();
+    	bfs.add(s);
+    	//while there are still nodes on the tree
+    	while(!bfs.isEmpty()){
```

Comment:
I like how you commented each step here; it's really easy to understand. also I think optimizing the S traversal with BFS via Queue is not only really elegant, but also a great way to do it if the interviewer doesn't allow recursion, so good job there! :)